modules/home/transformations/generate-sidebar-upgrade-for-core-and-pro.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace elementor\modules\home\transformations;
+
+use Elementor\Modules\Home\Transformations\Base\Transformations_Abstract;
+use Elementor\Utils;
+
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+class Generate_Sidebar_Upgrade_For_Core_And_Pro extends Transformations_Abstract {
+	public bool $has_pro;
+
+	public function __construct( $args ) {
+		parent::__construct( $args );
+
+		$this->has_pro = Utils::has_pro();
+	}
+
+	public function transform( array $home_screen_data ): array {
+		foreach ( $home_screen_data['sidebar_upgrade'] as $index => $item ) {
+			if ( $this->has_pro && 'free' === $item['license'][0] ) {
+				unset( $home_screen_data['sidebar_upgrade'][ $index ] );
```

Comment:
maybe it will be easier just generate a new array which will replace `sidebar_upgrade`. Should be easy to read: 
```
foreach(...) {
$new_sidebar = ...
}
$home_screen_data['sidebar_upgrade'] = $new_sidebar;

return $home_screen_data;

```

Reply:
I like @mykytamurzin's suggestion. In that way we can also keep the old datastructure with: 
`sidebar_upgrade`
`sidebar_upgrade.header`
`sidebar_upgrade.repeater`

etc.

Reply:
@mykytamurzin & @hein-obox I'm not sure I understand, the final output to frontend will have the original structure.

Reply:
it's a chat gpt just to demonstrate the idea 

```
function transform(array $home_screen_data): array {
    $new_sidebar_upgrade = [];

    foreach ($home_screen_data['sidebar_upgrade'] as $index => $item) {
        $has_pro_free_license = $this->has_pro && 'free' === $item['license'][0];
	$no_pro_empty_license = !$this->has_pro && 1 < count($item['license']);
	$should_not_show = isset($item['show']) && 'false' === $item['show'];
	
        if ($has_pro_free_license || $no_pro_empty_license || $should_not_show) {
            continue;
        }

        $new_sidebar_upgrade[] = $item;
    }

    $home_screen_data['sidebar_upgrade'] = reset($new_sidebar_upgrade);

    return $home_screen_data;
}
```

Reply:
looks similar to @TzviRabinovitch suggestion